### PR TITLE
various minor SMB and unionmount-testsuite changes

### PR DESCRIPTION
**Pull request by ddiss**
```
The following changes since commit cf361da7f1ba028ea8e03fc4430c0d8ecd7f4c84:

  Merge remote-tracking branch 'ddiss/ntfs3' (2023-03-24 00:16:42 +0100)

are available in the Git repository at:

  https://github.com/ddiss/rapido/ smb_ovl_various

for you to fetch changes up to 09ffb601113031fd379c9163b54e92a58755a890:

  unionmount_testsuite: save test output to file (2023-04-19 16:01:10 +0200)

----------------------------------------------------------------
David Disseldorp (6):
      runtime.vars: call _fail() instead of _fatal()
      autorun/ksmbd-btrfs: handle /usr/local prefixed paths
      samba-local: configure Btrfs backed shares
      kernel: enable CONFIG_CIFS_DFS_UPCALL
      unionmount-testsuite: fix python deps and remove shutdown
      unionmount_testsuite: save test output to file

 autorun/ksmbd_btrfs.sh          |  4 +++-
 autorun/samba_local.sh          | 23 +++++++----------------
 autorun/unionmount_testsuite.sh | 15 ++-------------
 cut/cifs.sh                     |  2 +-
 cut/fstests_cifs.sh             |  2 +-
 cut/samba_local.sh              |  2 +-
 cut/unionmount_testsuite.sh     | 26 +++++++++++---------------
 kernel/sle12sp3_config          |  2 +-
 kernel/sle15_config             |  2 +-
 kernel/sle15sp1_config          |  2 +-
 kernel/sle15sp2_config          |  2 +-
 kernel/sle15sp3_config          |  2 +-
 kernel/sle15sp4_config          |  2 +-
 kernel/vanilla_config           |  2 +-
 runtime.vars                    |  2 +-
 15 files changed, 34 insertions(+), 56 deletions(-)
```

### CODE REVIEW

File: cut/unionmount_testsuite.sh [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
-# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
-# License for more details.
 
 RAPIDO_DIR="$(realpath -e ${0%/*})/.."
 . "${RAPIDO_DIR}/runtime.vars"
 
 _rt_require_dracut_args "$RAPIDO_DIR/autorun/unionmount_testsuite.sh" "$@"
 _rt_require_conf_dir UNIONMOUNT_TESTSUITE_SRC
-_rt_mem_resources_set "1024M"
+_rt_mem_resources_set "4096M"
```
Comment (pevik) [inline cut/unionmount_testsuite.sh:10]: Quite big RAM, but I suppose it's necessary.

Comment (ddiss) [reply cut/unionmount_testsuite.sh:10]: As we're dumping all python paths in the image, it depends on how many python packages are installed :-)
On my tw dev box it was around 500M (perhaps mostly libvirt?), and I was hitting oom with 1G. I considered doing some fancy calculations with du, but figured I'd keep it simple for now.

File: runtime.vars [hunk 2]
```diff
@@ -203,7 +203,7 @@ _rt_require_samba_srv() {
 	shift
 	req_inst_ref+=("${mods_p}/pdb/tdbsam.so")
 	while (( $# > 0 )) ; do
-		[ -f "${mods_p}/${1}" ] || _fatal "Samba module $1 missing"
+		[ -f "${mods_p}/${1}" ] || _fail "Samba module $1 missing"
```
Comment (pevik) [inline runtime.vars:206]: I'd add to the commit message:
```
Fixes: 525e12b ("samba: support use of locally installed samba binaries")
```

Comment (ddiss) [reply runtime.vars:206]: good idea, will do

File: autorun/ksmbd_btrfs.sh [hunk 3]
```diff
@@ -27,7 +27,9 @@ echo "${CIFS_USER}:x:${cifs_xid}:" >> /etc/group
 echo "${FSTESTS_ZRAM_SIZE:-1G}" > /sys/block/zram0/disksize \
 	|| _fatal "failed to set zram disksize"
 mkfs.btrfs /dev/zram0 || _fatal "mkfs failed"
-mkdir -p /mnt/ /etc/ksmbd
+# ksmbd-tools may be built with a default prefix "/usr/local" or "/". There's
+# no easy way to find out, so account for both paths.
+mkdir -p /mnt/ /etc/ksmbd /usr/local/etc/ksmbd /usr/local/var/run/
```
Comment (pevik) [inline autorun/ksmbd_btrfs.sh:32]: very nit: I'd remove trailing slashes (obviously all are directories)

**Comment by pevik** [comment]
I consider loosing XFS in "samba-local: configure Btrfs backed shares" as not optimal (IMHO support was there quite long time, but ack (I suppose nobody will complain or just stick with older git version).

**Review by pevik — APPROVED** [review]

**Comment by ddiss** [comment]
> I consider loosing XFS in "samba-local: configure Btrfs backed shares" as not optimal (IMHO support was there quite long time, but ack (I suppose nobody will complain or just stick with older git version).

Fair enough. In that case I'll add a new samba-xfs runner, as proposed in the commit message.

**Comment by ddiss** [comment]
> > I consider loosing XFS in "samba-local: configure Btrfs backed shares" as not optimal (IMHO support was there quite long time, but ack (I suppose nobody will complain or just stick with older git version).
> 
> Fair enough. In that case I'll add a new samba-xfs runner, as proposed in the commit message. Thanks a lot for the review Petr!

I've done this and renamed `samba-local` to `samba-btrfs` (with an alias for the old name via cut symlink). All of your other points should also be addressed in this version.

**Pull request merged**